### PR TITLE
[enhancement](stats) Retry when loading stats

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

Retry at most to load stats when nereids enabled

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         32 seconds loaded 2358488459 Bytes, about 70 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.4 seconds inserted 10000000 Rows, about 352K ops/s
 storage size: 17169355115 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.2 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          39 seconds loaded 861443392 Bytes, about 21 MB/s
 insert into select:          30.1 seconds inserted 10000000 Rows, about 332K ops/s
 storage size: 17162499601 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.61 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.8 seconds inserted 10000000 Rows, about 335K ops/s
 storage size: 17168737591 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java [hunk 1]
```diff
@@ -67,4 +82,43 @@ private Optional<ColumnStatistic> loadFromStatsTable(long tableId, long idxId, S
...
+            this.retryTimes = retryTimes;
+        }
+
+        @Override
+        public void run() {
+            List<ResultRow> columnResults = null;
+            try {
+                columnResults = StatisticsRepository.loadColStats(key.tableId, key.idxId, key.colName);
+            } catch (InternalQueryExecutionException e) {
+                if (this.retryTimes < StatisticConstants.LOAD_RETRY_TIMES) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java:106]: what will happen if retryTimes >= LOAD_RETRY_TIMES? why not retry until load success or detected key aready been loaded into cache?

File: fe/fe-core/src/main/java/org/apache/doris/common/ThreadPoolManager.java [hunk 2]
```diff
@@ -134,6 +134,15 @@ public static ThreadPoolExecutor newDaemonFixedThreadPool(int numThread, int que
                 poolName, needRegisterMetric);
     }
 
+    public static ThreadPoolExecutor newDaemonFixedThreadPool(int numThread, int queueSize,
+                                                              String poolName, int timeoutSeconds,
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/common/ThreadPoolManager.java:138]: unuse arg `timeoutSeconds`

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java [hunk 3]
```diff
@@ -67,4 +82,43 @@ private Optional<ColumnStatistic> loadFromStatsTable(long tableId, long idxId, S
             return Optional.of(columnStatistics);
         }
     }
+
+    private void retryLoad(StatisticsCacheKey key) {
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java:87]: remove this blank line

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatisticsCacheLoader.java:106]: 能用于做重试的资源是有限的，无限重试有可能导致需要retry的stats越堆越多

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.87 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17160850363 Bytes

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/common/ThreadPoolManager.java [hunk 4]
```diff
@@ -134,6 +134,15 @@ public static ThreadPoolExecutor newDaemonFixedThreadPool(int numThread, int que
                 poolName, needRegisterMetric);
     }
 
+    public static ThreadPoolExecutor newDaemonFixedThreadPool(int numThread, int queueSize,
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/common/ThreadPoolManager.java:137]: could you explain the difference between newDaemonFixedThreadPool and newDaemonFixedPriorityThreadPool? what is been "fixed"

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/common/ThreadPoolManager.java:137]: thread number

**Review by englefly — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.21 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17169447222 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.6 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162675520 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.14 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162195194 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**